examples/server/daily-bot-manager.py

```diff
@@ -1,101 +1,165 @@
-            ),
-            500,
-        )
+        raise Exception(f"Unable to create room: {res.text}")
+
     room_url = res.json()["url"]
     room_name = res.json()["name"]
+    return (room_url, room_name)
 
-    meeting_token = get_meeting_token(room_name, daily_api_key, exp)
 
-    if args:
-        extra_args = " ".join([f'-{x[0]} "{x[1]}"' for x in args])
-    else:
-        extra_args = ""
+def create_token(room_name, token_properties, exp):
+    token_props = {"exp": exp, "is_owner": True}
+    if token_properties:
+        token_props |= token_properties
+    # Force the token to be limited to the room
+    token_props |= {"room_name": room_name}
+    res = requests.post(
+        f'{api_path}/meeting-tokens',
+        headers={
+            'Authorization': f'Bearer {daily_api_key}'},
+        json={
+            'properties': token_props})
+    if res.status_code != 200:
+        if res.status_code != 200:
+            raise Exception(f"Unable to create meeting token: {res.text}")
+
+    meeting_token = res.json()['token']
+    return meeting_token
+
+
+def start_bot(*, bot_path, room_url, token, bot_args, wait_for_bot):
 
+    room_name = get_room_name(room_url)
     proc = subprocess.Popen(
-        [f"python {bot_path} -u {room_url} -t {meeting_token} -k {daily_api_key} {extra_args}"],
+        [f"python {bot_path} -u {room_url} -t {token} -k {daily_api_key} {bot_args}"],
         shell=True,
         bufsize=1,
     )
 
-    # Don't return until the bot has joined the room, but wait for at most 2
-    # seconds.
-    attempts = 0
-    while attempts < 20:
-        time.sleep(0.1)
-        attempts += 1
-        res = requests.get(
-            f"{api_path}/rooms/{room_name}/get-session-data",
-            headers={"Authorization": f"Bearer {daily_api_key}"},
-        )
-        if res.status_code == 200:
-            break
-    print(f"Took {attempts} attempts to join room {room_name}")
-
-    # Additional client config
-    config = {}
-    if os.getenv("CLIENT_VAD_TIMEOUT_SEC"):
-        config['vad_timeout_sec'] = float(
-            os.getenv("DAILY_CLIENT_VAD_TIMEOUT_SEC"))
+    if wait_for_bot:
+        # Don't return until the bot has joined the room, but wait for at most 5
+        # seconds.
+        attempts = 0
+        while attempts < 50:
+            time.sleep(0.1)
+            attempts += 1
+            res = requests.get(
+                f"{api_path}/rooms/{room_name}/get-session-data",
+                headers={"Authorization": f"Bearer {daily_api_key}"},
+            )
+            if res.status_code == 200:
+                print(f"Took {attempts} attempts to join room {room_name}")
+                return True
+
+        # If we don't break from the loop, that means we never found the bot in the room
+        raise Exception("The bot was unable to join the room. Please try again.")
+
+    return True
+
+
+@app.route("/start/<string:text>", methods=["GET", "POST"])
+def start(text):
+    if text in APPS:
```

Comment:
nitpick: might be easier to read (or at least less indentation) if you check `if text not in APPS` and then immediately raise an exception.